src/app/OperationalSessionSetup.cpp

```diff
@@ -420,6 +435,28 @@ void OperationalSessionSetup::PerformAddressUpdate()
 
 void OperationalSessionSetup::OnNodeAddressResolved(const PeerId & peerId, const ResolveResult & result)
 {
+    // Loop over all the addresses that has been found and add them to the pool of potential address to try
+    // to connect to. The "best" result is not stored into the pool as it will be always tried first.
+    for (size_t i = 0; i < result.numIPs; i++)
+    {
+        auto & currentAddress = result.ipAddress[i];
+        if (currentAddress == result.address.GetIPAddress())
+        {
+            continue;
+        }
+
+        ResolveResult * resolveResult = mDiscoveredResults.CreateObject();
+        if (resolveResult == nullptr)
+        {
+            // It may happens if there are more results than the available number of slots in the pool. This
+            // is not fatal but it may indicate that the pool size must be increase.
```

Comment:
```suggestion
            // It may happen if there are more results than the available number of slots in the pool. This
            // is not fatal but it may indicate that the pool size must be increased.
```